### DISCUSSION TITLE
Add a primaryColorShade theme property

### DISCUSSION THREAD

**Discussion by empz** [mantinedev/mantine#6599] [Feature requests]
When adding a new color using the [generator](https://mantine.dev/colors-generator/), the tool will automatically decide where in the shade the base color is. That's okay, but it also means that a button primary will have the base color as the primary, but a different shade.

Take this color for example: https://mantine.dev/colors-generator/?color=983fff

![image](https://github.com/user-attachments/assets/40b26589-c386-4fdf-ba8c-2757201a491a)

I want the button (and every other component that has a color) to use this primary color, but the shade 4. Setting `primaryShade: 4` in the theme changes the default shade for ALL colors, which is very inconvenient. I thought the primaryShade would only affect the primaryColor, but that's not the case. I think there should be a way to change the primaryShade only for the primaryColor. Maybe a new theme prop like `primaryColorShade`.

**Comment by haltcase** [comment]
It would be great to be able to specify primary shades for each individual color, defaulting to the global `primaryShade` if unspecified. Some of my colors are just too light or dark to put them at the same index as our primary color's default shade, and now I have to remember to specify the shade for those colors every single time I use them, e.g., `color.5`.
